src/auth.rs

```diff
@@ -87,3 +90,171 @@ impl AppAuth {
         create_jwt(self.app_id, &self.key).context(crate::error::JWTSnafu)
     }
 }
+
+/// The data necessary to authenticate as a github OAtuh app.
+#[derive(Clone, Deserialize)]
+#[serde(from = "OAuthWire")]
+pub struct OAuth {
+    pub access_token: SecretString,
+    pub token_type: String,
+    pub scope: Vec<String>,
+}
+
+/// The wire format of the OAuth struct.
+#[derive(Deserialize)]
+struct OAuthWire {
+    access_token: String,
+    token_type: String,
+    scope: String,
+}
+
+impl From<OAuthWire> for OAuth {
+    fn from(value: OAuthWire) -> Self {
+        OAuth {
+            access_token: SecretString::from(value.access_token),
+            token_type: value.token_type,
+            scope: value.scope.split(",").map(ToString::to_string).collect(),
+        }
+    }
+}
+
+impl crate::Octocrab {
+    /// Authenticate with Github's device flow. This starts the process to obtain a new `OAuth`.
+    ///
+    /// See https://docs.github.com/en/developers/apps/building-oauth-apps/authorizing-oauth-apps#device-flow for details.
+    ///
+    /// Note: To authenticate against public Github, the `Octocrab` that calls this method
+    /// *must* be constructed with `base_url: "https://github.com"` and extra header
+    /// "ACCEPT: application/json". For example:
+    /// ```no_run
+    /// # async fn run() -> octocrab::Result<()> {
+    /// # use reqwest::header::ACCEPT;
+    /// let crab = octocrab::Octocrab::builder()
+    /// .base_url("https://github.com")?
+    /// .add_header(ACCEPT, "application/json".to_string())
+    /// .build()?;
+    /// # Ok(())
+    /// # }
+    /// ```
```

Comment:
I feel like the default should just work here. Right now this always fails:
```rust
let crab = Octocrab::default();
crab.authenticate_as_device(client_id, scopes).await?;
```

I wish there was a way to make the default just work. One idea is introspecting `self` and correcting missing accept headers and the default base url (which will never work). We wouldn't do anything if the user overwrite the base url. What do you think?